### ISSUE TITLE
Cannot use phpseclib\Net\SFTP as SFTP because the name is already in use in ../knplabs/gaufrette/src/Gaufrette/Adapter/PhpseclibSftp.php on line 6

### ISSUE THREAD

**Issue by playr1983g** [KnpLabs/Gaufrette#394]
I get the following error when using Gaufrette in a Symfony 2.4 environment when i do a cac:clear:

  `Cannot use phpseclib\Net\SFTP as SFTP because the name is already in use in ../knplabs/gaufrette/src/Gaufrette/Adapter/PhpseclibSftp.php on line 6`

I'm using the following versions:

knplabs/gaufrette | 0.2.1
knplabs/knp-gaufrette-bundle | 0.2.0

**Comment by akovalyov** [comment]
We are talking about https://github.com/KnpLabs/Gaufrette/blob/0.2.1/src/Gaufrette/Adapter/PhpseclibSftp.php#L6, right?
Do you have `ssh2` extension installed?

**Comment by playr1983g** [comment]
@akovalyov yes, that use statement.

I removed that **use** statement and  changed the constructor parameters from:

`public function __construct(SFTP $sftp,...)`

to 

`public function __construct(Sftp $sftp, ...)`

and it worked for me for the moment.
I have no ssh2 extension installed. My PHP version is PHP 5.5.9-1ubuntu4.14
I also tried to install the ssh2 extension - if needed - with no effect..

http://www.codexpedia.com/apache-server/install-phps-ssh2-extension-on-ubuntu-and-mac/

**Comment by akovalyov** [comment]
@playr1983g thanks for your report.

well, I am not quite sure from where does this issue come. Which version of `phpseclib` do you have? Could you list your PHP modules with `php -m`?

Thanks in advance.

**Comment by akovalyov** [comment]
@playr1983g just forgot to clarify - no, `ssh2` extension is not needed. I was just asking that because according to your error message, there is already a global class with `SFTP` name and you have a name collision there, so I just was thinking about that.
PHPSecLib means pure php impl of several protocols, there is no need of ssh2 extension there.

**Comment by playr1983g** [comment]
Ok thanks @akovalyov 

phpseclib:

`phpseclib/phpseclib                  2.0.1`

This my `php -m` output  (now of course with ssh2 included because i installed it :) ) 

```
[PHP Modules]
apc
apcu
bcmath
bz2
calendar
Core
ctype
curl
date
dba
dom
ereg
exif
fileinfo
filter
ftp
gd
gettext
hash
iconv
intl
json
libxml
mbstring
mhash
mysql
mysqli
openssl
pcntl
pcre
PDO
pdo_mysql
Phar
posix
readline
Reflection
session
shmop
SimpleXML
soap
sockets
SPL
ssh2
standard
sysvmsg
sysvsem
sysvshm
tokenizer
wddx
xdebug
xml
xmlreader
xmlwriter
Zend OPcache
zip
zlib

[Zend Modules]
Xdebug
Zend OPcache


```

**Comment by playr1983g** [comment]
Hi, my colleague had the same error...
We had `gaufrette 0.2.0` installed (working fine)
After a composer update we got a new minor `0.2.1` which introduced this error..

**Comment by NiR-** [comment]
@playr1983g Can you provide the complete stack trace please ?

**Comment by playr1983g** [comment]
Hi @NiR- 

```
PHP Fatal error:  Cannot use phpseclib\Net\SFTP as SFTP because the name is already in use in .../projects/ourproject/vendor/knplabs/gaufrette/src/Gaufrette/Adapter/PhpseclibSftp.php on line 6
PHP Stack trace:
PHP   1. {main}() .../projects/ourproject/app/console:0
PHP   2. Symfony\Component\Console\Application->run() .../projects/ourproject/app/console:27
PHP   3. Symfony\Bundle\FrameworkBundle\Console\Application->doRun() .../projects/ourproject/vendor/symfony/symfony/src/Symfony/Component/Console/Application.php:121
PHP   4. Symfony\Component\HttpKernel\Kernel->boot() .../projects/ourproject/vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Console/Application.php:70
PHP   5. Symfony\Component\HttpKernel\Kernel->initializeContainer() .../projects/ourproject/app/bootstrap.php.cache:2303
PHP   6. Symfony\Component\DependencyInjection\ContainerBuilder->compile() .../projects/ourproject/app/bootstrap.php.cache:2524
PHP   7. Symfony\Component\DependencyInjection\Compiler\Compiler->compile() .../projects/ourproject/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/ContainerBuilder.php:619
PHP   8. OurVendor\Bundle\FrameworkBundle\DependencyInjection\Compiler\AwareCompilerPass->process() .../projects/ourproject/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/Compiler/Compiler.php:117
PHP   9. class_exists() .../projects/ourproject/vendor/our-vendor/framework-bundle/OurVendor/Bundle/FrameworkBundle/DependencyInjection/Compiler/AwareCompilerPass.php:27
PHP  10. spl_autoload_call() .../projects/ourproject/vendor/our-vendor/framework-bundle/OurVendor/Bundle/FrameworkBundle/DependencyInjection/Compiler/AwareCompilerPass.php:27
PHP  11. Symfony\Component\Debug\DebugClassLoader->loadClass() .../projects/ourproject/vendor/OurVendor/framework-bundle/OurVendor/Bundle/FrameworkBundle/DependencyInjection/Compiler/AwareCompilerPass.php:0

```
The difference compared to a default symfony2 project is a CompilerPass we use to inject some dependencies. In that code we loop over the container definitions.
On of them is apparently that Gaufrette class which breaks when trying to autoload it..

**Comment by RyanMcEvoy** [comment]
I can report that I am also having this issue, I just installed symplify/modular-routing which has a compiler pass and I receive exactly the same issue.

To recreate this issue should be as simple as
```
composer require symplify/modular-routing
```

```php
class AppKernel extends Kernel
{
    public function registerBundles()
    {
        $bundles = [
            // ...
            new Symfony\Cmf\Bundle\RoutingBundle\CmfRoutingBundle(),
            new Symplify\ModularRouting\SymplifyModularRoutingBundle(),
            // ...
        ];
    }
}
```

**Comment by stof** [comment]
@NiR- the issue is that there is already a SFTP class in the adapter namespace in Gaufrette. If the class is already loaded, the above error will happen when using the use statement. Use statements will conflict with classes existing in the target namespace in PHP (when they are defined)

**Comment by RyanMcEvoy** [comment]
I've put in a PR to fix this. Did a bit of digging and [Scrutinizer picked up on this.](https://scrutinizer-ci.com/g/KnpLabs/Gaufrette/issues/master/files/src/Gaufrette/Adapter/PhpseclibSftp.php?selectedLabels%5B0%5D=9&orderField=path&order=asc&honorSelectedPaths=0)

**Comment by akovalyov** [comment]
Closed by #399.
